### PR TITLE
[FEAT] user repos fetch ui piece implemented.

### PR DESCRIPTION
**Pull request by YASH-RAJ-HANS**

### CODE REVIEW

**Review by sahilrohera10 — APPROVED** [review]

File: client/my-app/src/app/repositories/page.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const access_token = localStorage.getItem("access_token");
+    const fetchRepos = async () => {
+      if (!accessToken || !username) {
+        // window.location.href = "http://localhost:3000";
+        // return;
+      }
```
Comment (yash-25log) [inline client/my-app/src/app/repositories/page.tsx:24]: why this?

File: client/my-app/src/app/repositories/page.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  if (isLoading) {
+    return <div className="text-center text-gray-600">Loading...</div>;
+  }
+
+  if (error) {
+    return <div className="text-center text-red-500">{error}</div>;
+  }
+
+  return (
+    <div className=" bg-gray-900 flex flex-col items-center py-8 px-4 o">
```
Comment (yash-25log) [inline client/my-app/src/app/repositories/page.tsx:74]: css issue?

File: client/my-app/src/app/repositories/page.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+interface RepositoriesProps {
+  accessToken?: string;
+  username?: string;
+}
+
+const Repositories: React.FC<RepositoriesProps> = ({
+  accessToken = "",
+  username = "YASH-RAJ-HANS",
+}) => {
```
Comment (yash-25log) [inline client/my-app/src/app/repositories/page.tsx:13]: passing props and simultaneously retrieving from local storage can cause confusion

File: client/my-app/src/app/repositories/page.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  const [repos, setRepos] = useState<any[]>([]);
```
Comment (yash-25log) [inline client/my-app/src/app/repositories/page.tsx:14]: should have a specific type to improve type safety

**Pull request merged**